packages/shared/src/browser.ts

```diff
@@ -64,17 +90,17 @@ export function isValidBrowser(): boolean {
  * @returns {boolean}
  */
 export function isBrowserOnline(): boolean {
-  const navigator = window?.navigator;
-  if (!inBrowser() || !navigator) {
+  const navigator = browser.window.navigator;
+
+  if (!navigator) {
```

Comment:
Ditto. 
This refactor for these two cases might be fine, however I don't think we can simply replace the usage of the `inBrowser()` helper with the new proxy-based approach as the two are not equivalent.

Reply:
Thinking about this a bit more, if this piece of code runs in an expo app, instead of returning false, its going to throw and stop app execution upon accessing `browser.window`